plugins/katello/developers.md

```diff
@@ -229,37 +229,109 @@ Make sure to follow the instructions and update the docs to use the current rele
-done
-```
+Below is mostly a script, but not quite. Read the comments.
 
-Review the modules for releasing - we should be keeping them up to date when making incompatible changes, but do any need bumps of the major or minor version? If so, do so now.
+```sh
+# Review the the changes since the last tag
+tig $(git describe --abbrev=0 --tags)..HEAD
+
+# If it needs a major or minor version bump
+# Note: usually we already bumped it for patch releases but do check!
+bundle exec rake module:bump:major
+bundle exec rake module:bump:minor
+
+# Update CHANGELOG.md
+# See https://github.com/skywinder/github-changelog-generator#github-token on using a token to avoid ratelimits
+github_changelog_generator -u theforeman -p $(basename $PWD) -o new-changelog.md --future-release $(bundle exec rake module:version)
+
+# Now merge new-changelog.md into CHANGELOG.md - do note that it doesn't handle
+# branches so if anything was merged to a stable branch you should manually put
+# it in the correct release. This may also break badly on stable branches.
 
-Consider the changes that have been made to the module since the last release, do we need a new release?
-Use [blacksmith](https://github.com/maestrodev/puppet-blacksmith) to push to forge.
+# Create the release commit
+git add CHANGELOG.md
+VERSION=$(bundle exec rake module:version) && git commit -m "Release $VERSION"
 
-```bash
-rake module:release
+# Create a new tag
+bundle exec rake module:tag
+
+# Only recently puppet-blacksmith learned to do GPG signed tags - check this
+git verify-tag $(bundle exec rake module:version)
+
+# If it failed you can do it manually
+# VERSION=$(bundle exec rake module:version) && git tag -s $VERSION -m "Version $VERSION"
+
+# Push the changes to github
+git push --follow-tags
+
+# Push the release to the forge
+# This assumes you have set the credentials previously
+bundle exec rake module:clean module:push
+
+# Automated using pass - https://www.passwordstore.org/
+# Assumes you have a katello/forge or theforeman/forge password configured
+BLACKSMITH_FORGE_USERNAME="$(jq -r .author metadata.json | tr [A-Z] [a-z])" && BLACKSMITH_FORGE_USERNAME=$BLACKSMITH_FORGE_USERNAME BLACKSMITH_FORGE_PASSWORD=$(pass show $BLACKSMITH_FORGE_USERNAME/forge | head -n 1) bundle exec rake module:push
+
+# Start the next minor version
+bundle exec rake module:bump_commit:patch
+
+# And push the new version
+git push
 ```
 
-Now, lock down Katello versions to the appropriate minor, e.g. if you release `puppet-katello 0.2.0` or the latest is 0.2.0, then in the Puppetfile
+### Step 7. Pin katello-installer puppet modules for release
 
+On the katello-installer's KATELLO-X.Y branch, we need to pin the modules to a minor release.
+
+For every module in `Puppetfile` pin it to a minor version. For example:
+
+```ruby
+mod 'katello/katello', '>= 7.0.0 < 7.1.0'
 ```
-mod 'katello',   '>= 0.2.0 < 0.3.0'
+
+The module `katello/katello_devel` has no releases and can be ignored.
```

Comment:
Should we even mention this as it'll be removed in https://github.com/Katello/katello-installer/pull/602?

Reply:
In 3.6 we actively removed it similar to that PR so the instructions might be not entirely correct here anyway. I'll just remove the comment for now.